src/spec_coerce/core.cljc

```diff
@@ -179,11 +179,35 @@
        (URI. x)
        x)))
 
+(defn type->sym [x]
+  (cond (int? x)     `integer?
+        (float? x)   `float?
+        (boolean? x) `boolean?   ;; pointless but valid
+        ;;(symbol? x)  `symbol?  ;; doesn't work.
+        ;;(ident? x)   `ident?   ;; doesn't work.
+        (string? x)  `string?
+        (keyword? x) `keyword?
+        (uuid? x)    `uuid?
+        (nil? x)     `nil?       ;; even more pointless but stil valid
+
+        ;;#?(:clj (uri? x))     #?(:clj `uri?) ;; doesn't work.
+        #?(:clj (decimal? x)) #?(:clj `decimal?)))
+
+(defn spec-is-set? [x]
+  "If the spec is given as a set, and every member of the set is the same type,
+  then we can infer a coercion from that shared type."
+  (let [infer-spec (try (eval x)
```

Comment:
`eval` is a problem, I guess you can instead use `read-string` to get the set value

Reply:
`eval` is for making the following tests pass:

```clj
::referenced-set
::calculated-set 
::nilable-referenced-set 
::nilable-calculated-set 
::nilable-referenced-set-kw
::nilable-calculated-set-kw
```
I'll try to see if there's another way

Reply:
Just to be clear, this is evaluating the spec, not the value. So this is not a vector for an injection attack as far as I'm aware.

Reply:
this project uses a solution for this, which is the `spec-infer`, you can use that. eval is a big no-no, even if not intended for injection it can be used to (imagine if you are reading some external user specs). Also, it doesn't work on cljs (unless you pull the bootstrapped, which requires having the entire compiler in the browser).

Reply:
Well, I don't think the aforementioned tests can be made to pass without `eval`, unless you have some other idea. I'm guessing the root of the problem is that clojure.spec is macro-based. Should I just remove or comment out those tests and revert the code to only care about explicitly given sets?

Reply:
> reading some external user specs

Since this is already code that is already being evaluated elsewhere, where does the danger come from? I'm not trying to be obstinate, just genuinely curious.

Reply:
this is more a feeling thing, I have had many issues with `eval` in other environments in the past, it's not something I'm willing to allow in, I fear many things I can't prevent, its a bazooka and should be avoided IMO. Also, there is the `cljs` issue, you can't do `eval` at runtime there without significant increase in bundle size (to include the cljs compiler). Did you check the `spec-inspect` thing? I use that to get the source spec in other situations here.

Reply:
https://github.com/wilkerlucio/spec-inspec/blob/master/src/com/wsscode/spec_inspec.cljc#L29-L38

Reply:
I did check `spec-inspect`, and it only gets as far as returning the symbols as written in the code. So, for example:

```clj
;; Specs given as code
(def enum-map {:foo "bar"
               :baz "qux"})
(s/def ::calculated-set (->> enum-map keys (into #{})))

(si/spec->root-sym ::calculated-set)
=> 
(clojure.core/->>
 spec-coerce.core-test/enum-map
 clojure.core/keys
 (clojure.core/into #{}))


;; Specs given as referential symbols
(def enum-set #{:a :b})
(s/def ::referenced-set enum-set)

(si/spec->root-sym ::referenced-set)
=>
user/enum-set

```

To get the actual sets in question out of there, the only way I know of is `eval`. Again, I blame clojure.spec and its naughty macros. Thankfully, it sounds like spec2 is moving away from macros.

I definitely understand the lack of trust concerning `eval`. "Bazooka" is an apt term. So if the answer is "no," I'll simplify the code and we can just mention the limitation in the README. Or, perhaps, we could provide an optional flag or dynamic var binding to turn the `eval` feature on, so that crazy people like me can take it at their own risk. 

Concerning cljs, we could simply use the `#?` reader conditional. 

Let me know which way to go and I'll submit the changes ASAP.

Reply:
Yeah, spec2 should be much better. I prefer to say no to`eval` and accept the limitation of it (the set must be literal, not a reference).